src/julia_gc.cc

```diff
@@ -138,30 +525,12 @@ static void JFinalizer(jl_value_t * obj)
         TabFreeFuncBags[tnum]((Bag)&contents);
 }
 
-#if !defined(SCAN_STACK_FOR_MPTRS_ONLY)
-
-/****************************************************************************
-**
-**  Treap functionality
-**
-**  Treaps are probabilistically balanced binary trees. We use them for
-**  range queries on pointers for conservative scans. Unlike red-black
-**  trees, they're simple to implement, and unlike AVL trees, insertions
-**  take an expected O(1) number of mutations to the tree, making them
-**  more cache-friendly for an insertion-heavy workload.
-**
-**  Their downside is that they are probabilistic and that hypothetically,
-**  degenerate cases can occur. However, these are very unlikely, and if
-**  that turns out to be a problem, we can replace them with alternate
-**  balanced trees (B-trees being a likely suitable candidate).
-*/
-
 // Comparing pointers in C without triggering undefined behavior
 // can be difficult. As the GC already assumes that the memory
 // range goes from 0 to 2^k-1 (region tables), we simply convert
 // to uintptr_t and compare those.
 
-static inline int cmp_ptr(void * p, void * q)
+int cmp_ptr(void * p, void * q)
```

Comment:
Why?

Reply:
That's a problem with C++98/C++03, which does not allow you to pass static functions as template parameters. Using them as template parameters allows inlining (as opposed to storing a function pointer and the overhead that comes with that). I'm not happy with it, either, and have been thinking about workarounds, such as using inheritance instead or classes that function as comparators. Or, alternatively, have a function with a more unique name.

Reply:
I've done some code reorganization and put potentially conflicting symbols in a namespace of their own. And `cmp_ptr` is back to being a static inline function regardless.